gen/target.cpp

```diff
@@ -64,9 +85,34 @@ llvm::Type *getRealType(const llvm::Triple &triple) {
   case Triple::wasm64:
     return LLType::getFP128Ty(ctx);
 
+  case Triple::ppc64:
+  case Triple::ppc64le:
+    if (realPrecision == RealPrecision::DoubleDouble) {
+      return LLType::getPPC_FP128Ty(ctx);
+    }
+    if (triple.isMusl()) { // Musl uses double
+      return LLType::getDoubleTy(ctx);
+    }
+#if defined(__linux__) && defined(__powerpc64__)
+    // for a PowerPC64 Linux build:
+    // default to the C++ host compiler's `long double` ABI
+    switch (std::numeric_limits<long double>::digits) {
+    case 113:
+      return LLType::getFP128Ty(ctx);
+    case 106:
+      return LLType::getPPC_FP128Ty(ctx);
+    case 53:
+      return LLType::getDoubleTy(ctx);
+    default:
+      llvm_unreachable("Unexpected host C++ 'long double' precision for a "
+                       "PowerPC64 target!");
+    }
```

Comment:
> perhaps print the actual __LDBL_MANT_DIG__ number in the diagnostic? (will help troubleshooting when this error triggers)

@JohanEngelen  Like this?

```suggestion
#if __LDBL_MANT_DIG__ == 113
      return LLType::getFP128Ty(ctx);
#elif  __LDBL_MANT_DIG__ == 106
      return LLType::getPPC_FP128Ty(ctx);
#elif __LDBL_MANT_DIG__ == 53
      return LLType::getDoubleTy(ctx);
#else
#error "Unexpected host C++ 'long double' precision for a PowerPC64 target!"
#endif
```

Reply:
if that works, then that's great yeah :)